projects/cds/src/profiletag/services/spartacus-event.service.ts

```diff
@@ -50,4 +64,22 @@ export class SpartacusEventService {
       }))
     );
   }
+
+  /**
+   * Listens to successful logins and pushes the event for profiletag to pick it up further
+   */
+  loginSuccessful(): Observable<boolean> {
+    return this.actionsSubject.pipe(
+      filter(
+        action =>
+          action.type === AuthActions.LOGIN ||
+          action.type === AuthActions.LOGOUT
```

Comment:
As you are filtering for `LOGIN` on line 81, not sure why do you have `LOGOUT` here?
If the reason is because this method won't fire in case the user logs out and then logs back in, then you can maybe try to remove the `distinctUntilChanged` and check if it works?

Reply:
@znikola exactly. It was the only way to get this test to pass: https://github.com/SAP/cloud-commerce-spartacus-storefront/blob/af1b940c27b43e6d915f0b820276993843381eef/projects/cds/src/profiletag/services/spartacus-event.service.spec.ts#L198
We were basically guarding here against a login happening twice, while allowing a logout and then login to still trigger it.
Maybe we do not need to guard against a login event happening twice here, as this would be a bug in the system. Should I remove this safeguard?

Reply:
It's up to you, but I don't think it is necessary to guard against this.
The login action will fire twice, if in the same session the user logs in -> logs out -> logs back in (with the same credentials or with other credentials).

Reply:
BTW, I meant to say `guarding here against a login happening twice in a row`

Reply:
I don't think you need to do that. It would be a serious bug if Login action was fired twice in a row :)